lib/web_ui/flutter_js/src/utils.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+export const baseUri = getBaseURI();
+
+function getBaseURI() {
+  const base = document.querySelector("base");
+  return (base && base.getAttribute("href")) || "";
+}
+
+export function joinPathSegments(...segments) {
+  return segments.map((segment, i) => {
```

Comment:
I think this should just do the logical thing when handed any `string?`, and not throw. I don't want to add extra error handling to this function. I can pre-filter out falsey strings, but I still would need to do the `.length` filter afterward in case a string was just a slash, and it only became empty after the map function.

Reply:
If we're using length to validate, maybe also "trim" beforehand, so "     " is not a valid string?